udata/tests/api/test_datasets_api.py

```diff
@@ -47,6 +47,44 @@ def test_dataset_api_list(self):
         self.assertEqual(len(response.json['data']), len(datasets))
         self.assertFalse('quality' in response.json['data'][0])
 
+    def test_dataset_api_list_filtered_by_org(self):
+        '''It should fetch a dataset list for a given org'''
+        self.login()
+        with self.autoindex():
+            member = Member(user=self.user, role='editor')
+            org = OrganizationFactory(members=[member])
+            DatasetFactory(resources=[ResourceFactory()])
```

Comment:
You can also use `VisibleDatasetFactory`: it prevent manual resource creation and if the visibility rule evolve the test stay valid.